lib/src/json_schema/utils.dart

```diff
@@ -106,3 +106,20 @@ class DefaultValidators {
     }
   }
 }
+
+extension DeepMerge<K> on Map<K, dynamic> {
+  /// Merge other into this with properties in this taking precedence.
+  /// Lists ARE NOT concatenated together.
+  Map<K, dynamic> deepMerge(Map<K, dynamic> other) {
+    other.forEach((key, value) {
+      if (this.containsKey(key)) {
+        if (value is Map && this[key].runtimeType == value.runtimeType) {
+          this[key] = (this[key] as Map).deepMerge(value);
+        } // Else don't clobber what is is there
```

Comment:
Chatted live - unfortunately [the spec doesn't say](https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.8.2.4) what to do if the applied schema that has keywords that collide with what is local. But this behavior makes sense to us :man_shrugging: